### ISSUE TITLE
Please import only the functions you use from data.table

### ISSUE THREAD

**Issue by tdhock** [Carm1r/fruclimadapt#1]
hi @Carm1r 
There are several instances of `@import data.table` in your roxygen comments, which get converted to `import(data.table)` in your NAMESPACE, which has the effect of importing everything that data.table exports.
In the new version of data.table on github master, `remotes::install_github("Rdatatable/data.table")` there are new exported objects `yearmon` and `yearqtr`.
There are also exported objects with these same two names in package zoo which you also import.
This has the effect of generating the following new warning when I run R CMD check fruclimadapt with data.table github master installed:
```
* checking whether package 'fruclimadapt' can be installed ... WARNING
Found the following significant warnings:
  Warning: replacing previous import 'data.table::yearmon' by 'zoo::yearmon' when loading 'fruclimadapt'
  Warning: replacing previous import 'data.table::yearqtr' by 'zoo::yearqtr' when loading 'fruclimadapt'
```
The fix would be to use `@importFrom data.table setDT` etc (for only the objects you use), instead of `@import data.table`.
Please update your code, then run R CMD check fruclimadapt_vers.tar.gz with data.table master installed.
After you update your code and R CMD check has no more errors, please submit a new version of fruclimadapt to CRAN.
This will help facilitate releasing a new version of data.table to CRAN.
(data.table devs must make sure all reverse dependencies do not break, before submitting a new version to CRAN)

**Comment by tdhock** [comment]
hi @Carm1r  Are you working on a fix? When should we expect an update on CRAN?
Thanks
Toby

**Comment by Carm1r** [comment]
Hi! Yes, I'm into fixing it and some other problems that have arisen in the last week. I expect having them solved in ten days tops.

**Comment by tdhock** [comment]
thanks!

**Comment by Carm1r** [comment]
Hi! The fixed version (0.4.5) is already uploaded on CRAN.

**Comment by tdhock** [comment]
I confirm that the new version on CRAN fixes our issue, thanks!
